tests/Testcontainers.Tests/Fixtures/Containers/Unix/Modules/Databases/CosmosDbFixture.cs

```diff
@@ -0,0 +1,58 @@
+namespace DotNet.Testcontainers.Tests.Fixtures
+{
+  using System;
+  using System.Data.Common;
+  using System.IO;
+  using System.Threading;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Builders;
+  using DotNet.Testcontainers.Configurations;
+  using DotNet.Testcontainers.Containers;
+
+  public class CosmosDbFixture : DatabaseFixture<CosmosDbTestcontainer, DbConnection>
+  {
+    public CosmosDbFixture()
+      : this(new CosmosDbTestcontainerConfiguration())
+    {
+    }
+
+    private CosmosDbFixture(CosmosDbTestcontainerConfiguration configuration)
+    {
+      this.Configuration = configuration;
+      this.Container = new TestcontainersBuilder<CosmosDbTestcontainer>()
+        .WithCosmosDb(configuration)
+        .Build();
+    }
+
+    public CosmosDbTestcontainerConfiguration Configuration { get; set; }
+
+    public override async Task InitializeAsync()
+    {
+      // workaround for broken cosmosdb emulator
+      var maxWait = TimeSpan.FromSeconds(5 * 1000);
```

Comment:
You do not need all that. `CancellationTokenSource` does that already:

```csharp
[UsedImplicitly]
public sealed class CosmosDbFixture : DatabaseFixture<CosmosDbTestcontainer, CosmosClient>
{
  private readonly CosmosDbTestcontainerConfiguration configuration = new CosmosDbTestcontainerConfiguration();

  private readonly CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));

  public CosmosDbFixture()
  {
    this.Container = new TestcontainersBuilder<CosmosDbTestcontainer>()
      .WithCosmosDb(this.configuration)
      .Build();
  }

  public override Task InitializeAsync()
  {
    return this.Container.StartAsync(this.cts.Token);
  }

  public override Task DisposeAsync()
  {
    return this.Container.DisposeAsync().AsTask();
  }

  public override void Dispose()
  {
    this.configuration.Dispose();
  }
}
```

Reply:
Didn't know that. You always learn something new :-)